src/odemis/driver/actuator.py

```diff
@@ -940,6 +940,188 @@ def reference(self, axes):
         return f
 
 
+class LinearActuator(model.Actuator):
+    """
+    A generic actuator component which allows moving on a linear axis. It is actually a
+    wrapper to just one axis/actuator. The actuator is referenced after n moves (can
+    be specified in constructor).
+    """
+
+    def __init__(self, name, role, children, axis_name, offset=0,
+                 ref_start=None, ref_period=10, inverted=None, **kwargs):
+        """
+        name (string)
+        role (string)
+        children (dict str -> actuator): axis name (in this actuator) -> actuator to be used for this axis
+        axis_name (str): axis name in the child actuator
+        offset (float): axis offset (negative of reference position value)
+        ref_start (float or None): Value usually chosen close to reference switch from where to start
+         referencing. Used to optimize runtime for referencing. If None, value will be 5% of value of cycle.
+        ref_period (int or None): number of moves before referencing axis, None to disable
+         automatic referencing
+        """
+        if inverted:
+            raise ValueError("Axes shouldn't be inverted")
+
+        if len(children) != 1:
+            raise ValueError("LinearActuator needs precisely one child")
+
+        axis, child = children.items()[0]
+        self._axis = axis
+        self._child = child
+        self._caxis = axis_name
+        self._offset = offset
+        self._ref_start = ref_start
+        self._ref_period = ref_period  # number of moves before automatic referencing
+        self._move_num = 0  # current number of moves after last referencing
+
+        # Executor used to reference and move to nearest position
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
+
+        if not isinstance(child, model.ComponentBase):
+            raise ValueError("Child %s is not a component." % (child,))
+        if not hasattr(child, "axes") or not isinstance(child.axes, dict):
+            raise ValueError("Child %s is not an actuator." % child.name)
+
+        ac = child.axes[axis_name]
+        rng = (ac.range[0] - self._offset, ac.range[1] - self._offset)
+        axes = {axis: model.Axis(range=rng, unit=ac.unit)}
+        model.Actuator.__init__(self, name, role, axes=axes, children=children, **kwargs)
+
+        # Offset from which to start referencing
+        if self.ref_start is None:
+            self.ref_start = abs(rng[1] - rng[0]) * 0.05
+        if not 0 <= ref_start <= abs(rng[1] - rng[0]):
```

Comment:
`ref_start` just needs to be between `rng[0]` and `rng[1]`. (And the error message needs an update)